[BUG] Fix flaky org.opensearch.knn.index.codec.jvector.KNNJVectorTests.testJVectorKnnIndex_simpleCase

```
][5EDB6F842A9D45]][org.opensearch.knn.index.codec.jvector.JVectorWriter]  Vector count: 1, less than limit to trigger PQ quantization: 1024, for field test_field, will use full precision vectors instead.
  2> java.lang.AssertionError: expected:<1> but was:<4>
        at __randomizedtesting.SeedInfo.seed([955EDB6F842A9D45:31573ABE682E7E07]:0)
        at org.junit.Assert.fail(Assert.java:89)
        at org.junit.Assert.failNotEquals(Assert.java:835)
        at org.junit.Assert.assertEquals(Assert.java:647)
        at org.junit.Assert.assertEquals(Assert.java:633)
        at org.opensearch.knn.index.codec.jvector.KNNJVectorTests.testJVectorKnnIndex_simpleCase(KNNJVectorTests.java:78)
```

Reproduce with: `./gradlew test --tests KNNJVectorTests.testJVectorKnnIndex_simpleCase -Dtests.seed=955EDB6F842A9D45`

See please https://github.com/opensearch-project/opensearch-jvector/actions/runs/23210133435/job/67462102190?pr=288

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -10,7 +10,6 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 * Enable derived sources for vectors to save storage costs. [241] (https://github.com/opensearch-project/opensearch-jvector/pull/241)
 ### Enhancements
 ### Bug Fixes
-* Fix flaky test KNNJVectorTests.testJVectorKnnIndex_simpleCase
 ### Infrastructure
 * Upgrade Lucene to 10.4.0 [292] (https://github.com/opensearch-project/opensearch-jvector/pull/292)
 ### Documentation

